src/common/htmlFileReader.ts

```diff
@@ -73,13 +75,42 @@ export default class HtmlFileReader {
             responseType: "blob",
         };
 
-        // Download the asset binary from the storage provider
-        const response = await axios.get<Blob>(asset.path, config);
-        if (response.status !== 200) {
-            throw new Error("Error downloading asset binary");
+        let response = null;
+        if (asset.type === AssetType.VideoFrame) {
+            response = await this.getAssetFrameImage(asset);
+        } else {
+            // Download the asset binary from the storage provider
+            response = await axios.get<Blob>(asset.path, config);
+            if (response.status !== 200) {
+                throw new Error("Error downloading asset binary");
+            }
         }
 
-        return response.data;
+        return await response.data;
+    }
+
+    public static async getAssetFrameImage(asset: IAsset) {
+        return new Promise<Blob>((resolve, reject) => {
+            const video = document.createElement("video");
+            const secs = asset.timestamp;
+            video.onloadedmetadata = function() {
+                this.currentTime = Math.min(Math.max(0, (secs < 0 ? this.duration : 0) + secs), this.duration);
+            }.bind(video);
+            video.onseeked = (e) => {
+                const canvas = document.createElement("canvas");
+                canvas.height = video.videoHeight;
+                canvas.width = video.videoWidth;
+                const ctx = canvas.getContext("2d");
+                ctx.drawImage(video, 0, 0, canvas.width, canvas.height);
+                canvas.toBlob((blob) => {
+                    resolve(blob);
+                });
+            };
+            video.onerror = (e) => {
+                reject(e);
+            };
+            video.src = asset.path;
```

Comment:
Even if the video object is created only on the virtual dom it seams to me that loading anytime the entire video files in memory for each individual frame we want to export could be an issue on both performance and memory consumption.  Just wonder if it is not better to implement a batch interface, load the video file once and return a list of all requested frames.

Reply:
So this is for the first pass still. I was hoping to see if the app was smart enough to do any caching of the data. If there are performance issue, which is definitely a possibility, it will be a batch process that just loads the video once.